timesketch/lib/graphs/interface.py

```diff
@@ -0,0 +1,217 @@
+# Copyright 2020 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Interface for graphs."""
+
+import hashlib
+
+from flask import current_app
+import networkx as nx
+
+from timesketch.lib.datastores.elastic import ElasticsearchDataStore
+
+
+GRAPH_TYPES = {
+    'Graph': nx.Graph,
+    'MultiGraph': nx.MultiGraph,
+    'DiGraph': nx.DiGraph,
+    'MultiDiGraph': nx.MultiDiGraph
+}
+
+
+class Graph(object):
+    """Graph object with helper methods.
+
+    Attributes:
+        nx_instance: Networkx graph object.
+    """
+    def __init__(self, graph_type):
+        """Initialize Graph object.
+
+        Args:
+            graph_type: (str) Name of graph type.
+        """
+        _nx_graph_class = GRAPH_TYPES.get(graph_type)
+        self.nx_instance = _nx_graph_class()
+        self._nodes = {}
+        self._edges = {}
+
+    def add_node(self, label, attributes):
+        """Add node to graph.
+
+        Args:
+            label: (str) Label for the node.
+            attributes: (dict) Attributes to add to node.
+
+        Returns:
+              Instance of Node object.
+        """
+        node = Node(label, attributes)
+        if node.id not in self._nodes:
+            self._nodes[node.id] = node
+        return node
+
+    def add_edge(self, source, target, label, event, attributes=None):
+        """Add edge to graph.
+
+        Args:
+            source: (Node) Node to use as source.
+            target: (Node) Node to use as target.
+            label: (str) Label for the node.
+            event: (dict): Elasticsearch event.
+            attributes: (dict) Attributes to add to node.
+        """
+        edge_id_string = ''.join([source.id, target.id, label]).lower()
+        edge_id = hashlib.md5(edge_id_string.encode('utf-8')).hexdigest()
+
+        edge = Edge(source, target, label, attributes)
+
+        if edge.counter < 500:
```

Comment:
why the 500? that sounds like a `magical` number that requires further explanation/comment